core/src/main/java/com/es/core/model/phone/JdbcPhoneDao.java

```diff
@@ -9,19 +8,42 @@
 import java.util.Optional;
 
 @Component
-public class JdbcPhoneDao implements PhoneDao{
+public class JdbcPhoneDao implements PhoneDao {
     @Resource
     private JdbcTemplate jdbcTemplate;
 
     public Optional<Phone> get(final Long key) {
-        throw new UnsupportedOperationException("TODO");
+        return jdbcTemplate.query("select ph.*, colors.code as color " +
+                        "from (select phones.* from phones where phones.id = ?) ph " +
+                        "left join phone2color on ph.id = phone2color.phoneId " +
+                        "left join colors on phone2color.colorId = colors.id",
+                new Object[]{key}, new PhoneRowMapper()).stream().findAny();
     }
 
     public void save(final Phone phone) {
-        throw new UnsupportedOperationException("TODO");
+        jdbcTemplate.update("INSERT INTO phones VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
+                phone.getId(), phone.getBrand(), phone.getModel(), phone.getPrice(), phone.getDisplaySizeInches(),
+                phone.getWeightGr(), phone.getLengthMm(), phone.getWidthMm(), phone.getHeightMm(),
+                phone.getAnnounced(), phone.getDeviceType(), phone.getOs(), phone.getDisplayResolution(),
+                phone.getPixelDensity(), phone.getDisplayTechnology(), phone.getBackCameraMegapixels(),
+                phone.getFrontCameraMegapixels(), phone.getRamGb(), phone.getInternalStorageGb(),
+                phone.getBatteryCapacityMah(), phone.getTalkTimeHours(), phone.getStandByTimeHours(),
+                phone.getBluetooth(), phone.getPositioning(), phone.getImageUrl(), phone.getDescription());
+
+        phone.getColors().stream()
+                .forEach(color -> {
+                    Long colorId = jdbcTemplate.queryForObject("SELECT colors.id FROM colors WHERE colors.code = '" + color.getCode() + "'",
+                            Long.class);
+                    jdbcTemplate.update("INSERT INTO phone2color VALUES (?, ?)", phone.getId(), colorId);
+                });
     }
 
     public List<Phone> findAll(int offset, int limit) {
-        return jdbcTemplate.query("select * from phones offset " + offset + " limit " + limit, new BeanPropertyRowMapper(Phone.class));
+        return jdbcTemplate.query("select ph.*, colors.code as color " +
+                        "from (select phones.* from phones offset ? limit ?) ph " +
+                        "left join phone2color on ph.id = phone2color.phoneId " +
+                        "left join colors on phone2color.colorId = colors.id",
+                new Object[]{offset, limit},
+                new PhoneRowMapper());
```

Comment:
Phone mapper should be injected with spring, not new PhoneRowMapper(), please consider this approach for all classes except model beans